packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-12/actions.test.js

```diff
@@ -0,0 +1,112 @@
+import test from 'ava';
+
+import {
+  adjustVault,
+  closeVault,
+  installBundle,
+  mintIST,
+  openVault,
+  proposalBuilder,
+  submitProposal,
+  voteLatestProposalAndWait,
+} from '../coreUpgradeHelpers.mjs';
+import { agd, agoric, agops } from '../cliHelper.mjs';
+import { GOV1ADDR, SDK_ROOT } from '../constants.mjs';
+import { installBundles, runZcfUpgrade, runProber } from './actions.mjs';
+
+test.before(async t => {
+  await mintIST(GOV1ADDR, 1234000000, 10000, 2000);
+
+  const bundlesData = [
+    {
+      name: 'Zcf-upgrade',
+      filePath: `${SDK_ROOT}/packages/zoe/src/contractFacet/vatRoot.js`,
+    },
+    {
+      name: 'Zoe-upgrade',
+      filePath: `${SDK_ROOT}/packages/vats/src/vat-zoe.js`,
+    },
+    {
+      name: 'prober-contract',
+      filePath: `${SDK_ROOT}/packages/boot/test/bootstrapTests/zcfProbe.js`,
+    },
+  ];
+
+  // @ts-expect-error
+  t.context.bundleIds = await installBundles(bundlesData);
+});
+
+test.serial('Open Vaults', async t => {
```

Comment:
important but for a follow-up PR:

the actions aren't test themselves. They're always in a strict order, which isn't really what Ava's `.serial` is meant for. (While it does execute in order, it's only guarantee is "not concurrently". Users should still be able to select any test in the file to run without the others.)

I think the _actions_ module should just be a script without any Ava tests. In this file I see the only assertions are strict equality, which can be achieved with [Node's built-in `assert()`](https://nodejs.org/api/assert.html)